update Networking page content (#3265)
bacalhau-project/bacalhau: docs/docs/setting-up/networking-instructions/networking.md

sidebar_label: "Internet Access"
---

> # Internet access

Reviewer: I wonder if we need to rewrite this more significantly. There's nothing fundamentally wrong with it, but it's framed in a weird way. You can use external docker containers, for example. It's just external network access that is limited - not necessarily to the internet, even to other nodes on the same network.

Further, the `--network` tag needs to be paired with a server with similar flags - do we need to detail that?

Author: Could you please provide more details for the node/server flag pairing? Seems this is not currently reflected in the documentation

Reviewer 2: If a user specifies the --network flag, then it will only be run on compute nodes that were started with the `--job-selection-accept-networked ` flag. If no compute nodes were started with that flag, the job will not run.

Author: Added a note for this case

By default, after launch Bacalhau jobs do not have any access to the external network - Internet or other nodes in the same network. This is to keep both compute providers and users safe from malicious activities.

However, there are 2 ways to access your data - before or during the execution of a job:
- Using Data Volumes to download the input data and upload the results
- Using `--network` flag to allow job to access the internet 
:::info
For more details about flags and recent updates see the [CLI Guide](../../dev/cli-reference/all-flags.md) or execute `bacalhau help`. Also feel free to contact us on [Slack](https://bacalhauproject.slack.com).
:::

## Using Data Volumes

> If you need to process data located on a certain network resource and/or save the result of the job execution to such resource - you will need to specify the internet locations to download data from and write results to when creating the job. Both [Docker](../../getting-started/docker-workload-onboarding.md) and [WebAssembly](../../getting-started/wasm-workload-onboarding.md) jobs support these features.

Reviewer 2: We might want to skip discussing saving the result of the job in this paragraph as it's mentioned in a paragraph below.  Perhaps this paragraph could just talk about downloading data?

Author: fixed

> When submitting a Bacalhau job, you can specify the CID (Content IDentifier) or HTTP(S) URL to download data from. The data will be retrieved before the job starts and made available to the job as a directory on the filesystem. When running Bacalhau jobs, you can specify as many CIDs or URLs as needed using `--input` which is accepted by both `bacalhau docker run` and `bacalhau wasm run`. See [command line flags](../../dev/cli-reference/all-flags.md) for more information. Make sure the nodes on the network have enough resources to download and process specified data.

Reviewer 2: ```suggestion
When submitting a Bacalhau job, you can specify the CID (Content IDentifier) or HTTP(S) URL to download data from. The data will be retrieved before the job starts and made available to the job as a directory on the filesystem. When running Bacalhau jobs, you can specify as many CIDs or URLs as needed using `--input` which is accepted by both `bacalhau docker run` and `bacalhau wasm run`. See [command line flags](../..all-flags.md) for more information. Make sure the nodes on the network have enough resources to download and process specified data.
```

> You can write back results from your Bacalhau jobs to your public storage location. By default, jobs will write results to the storage provider using the `--publisher` command line flag. See [command line flags](../../dev/cli-reference/all-flags.md) on how to configure this.

Reviewer 2: ```suggestion
You can write back results from your Bacalhau jobs to your public storage location. By default, jobs will write results to the storage provider using the `--publisher` command line flag. See [command line flags](../../all-flags.md) on how to configure this.
```

## Specifying Jobs to Access the Internet

> For some workloads, the required data is computed as part of the job if the purpose of the job is to process web results. In these cases, networking may be possible during job execution. To run Docker jobs on Bacalhau with internet access, you'll need to use a `--netwwork` flag and specify one of the following options:

Reviewer 2: ```suggestion
For some workloads there is a requirement for the job to fetch data at runtime, particularly when it requires access to external APIs. In these cases, networking may be possible during job execution. To run Docker jobs on Bacalhau with internet access, you'll need to use a `--netwwork` flag and specify one of the following options:
```

> ## Support for networked jobs on the public network

Reviewer: Let's stop referring to this as the "public" network - we should be talking about the "demo" network, we don't want the impression that we're running a public network for everyone for free.

Author: done
